contracts/crypto/SignatureChecker.sol

```diff
@@ -8,67 +8,103 @@ import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
 @notice Additional functions for EnumerableSet.Addresset that require a valid
-ECDSA signature, signed by any member of the set.
+ECDSA signature of a standardized message, signed by any member of the set.
  */
 library SignatureChecker {
     using EnumerableSet for EnumerableSet.AddressSet;
 
     /**
-    @notice Requires that the nonce has not been used previously and that the
+    @notice Requires that the message has not been used previously and that the
     recovered signer is contained in the signers AddressSet.
+    @dev Convenience wrapper for message generation + signature verification
+    + marking message as used
     @param signers Set of addresses from which signatures are accepted.
-    @param usedNonces Set of already-used nonces.
-    @param signature ECDSA signature of keccak256(abi.encodePacked(data,nonce)).
+    @param usedMessages Set of already-used messages.
+    @param signature ECDSA signature of message.
      */
     function validateSignature(
         EnumerableSet.AddressSet storage signers,
         bytes memory data,
-        bytes32 nonce,
         bytes calldata signature,
-        mapping(bytes32 => bool) storage usedNonces
+        mapping(bytes32 => bool) storage usedMessages
     ) internal {
-        require(!usedNonces[nonce], "SignatureChecker: Nonce already used");
-        usedNonces[nonce] = true;
-        _validate(signers, keccak256(abi.encodePacked(data, nonce)), signature);
+        bytes32 message = generateMessage(data);
+        require(
+            !usedMessages[message],
+            "SignatureChecker: Message already used"
+        );
+        usedMessages[message] = true;
+        requireValidSignature(signers, message, signature);
     }
 
     /**
-    @notice Requires that the recovered signer is contained in the signers
-    AddressSet.
-    */
+    @notice Requires that the message has not been used previously and that the
+    recovered signer is contained in the signers AddressSet.
+    @dev Convenience wrapper for message generation + signature verification.
+     */
     function validateSignature(
         EnumerableSet.AddressSet storage signers,
-        bytes32 hash,
+        bytes memory data,
         bytes calldata signature
     ) internal view {
-        _validate(signers, hash, signature);
+        bytes32 message = generateMessage(data);
+        requireValidSignature(signers, message, signature);
     }
 
     /**
-    @notice Hashes addr and requires that the recovered signer is contained in
-    the signers AddressSet.
-    @dev Equivalent to validate(sha3(addr), signature);
+    @notice Requires that the message has not been used previously and that the
+    recovered signer is contained in the signers AddressSet.
+    @dev Convenience wrapper for message generation from address +
+    signature verification.
      */
     function validateSignature(
         EnumerableSet.AddressSet storage signers,
-        address addr,
+        address address_,
```

Comment:
I really dislike this element of the style guide, but that's heavily influenced by my being used to Go's one. `<rant>`Given that the variable is used over only two lines and the type is `address` IMO it would even be acceptable to call it `a` instead of introducing a clunky underscore.`</rant>` :)

Reply:
I'm a bit of a burnt child with respect to variable naming. I had to work with some legacy code once where every variable had max 3 letters... That's why I'm a bit biased towards speaking (and sometimes admittedly overly verbose) variable names.
But no strong opinion in this context. I'll gladly change it.

Reply:
I think it's all about being thoughtful of the context… sometimes `addressOfSender` is appropriate and sometimes `a` is.